adapter.go

```diff
@@ -182,6 +198,37 @@ func NewAdapterByDBUseTableName(db *gorm.DB, prefix string, tableName string) (*
 	return a, nil
 }
 
+/*
+multiple databases support
+Example usage:
+	dbPool,err := initDbResolver([]gorm.Dialector{mysql.Open(dsn),mysql.Open(dsn2)},[]string{"casbin1","casbin2"})
+	a := initAdapterWithGormInstanceByMulDb(t,dbPool,"casbin1","","casbin_rule1")
+	a = initAdapterWithGormInstanceByMulDb(t,dbPool,"casbin2","","casbin_rule2")
+*/
+func initDbResolver(dbArr []gorm.Dialector, dbNames []string) (DbPool, error) {
+	if len(dbArr) == 0 {
+		panic("dbArr len is 0")
+	}
+	source, e := gorm.Open(dbArr[0])
+	if e != nil {
+		panic(e.Error())
+	}
+	p := specificPolicy{connnIndex: 0}
+	err := source.Use(dbresolver.Register(dbresolver.Config{Policy: &p, Sources: dbArr}))
+	dbMap := make(map[string]int, 3)
+	for i := 0; i < len(dbNames); i++ {
+		dbMap[dbNames[i]] = i
+	}
+	return DbPool{dbMap: dbMap, policy: &p, db: source}, err
+}
```

Comment:
This method is only used in test file, if it a testing support function, plz move it to test file.

Reply:
usage:
https://github.com/casbin/gorm-adapter/blob/a3ed926e413e04edce23b418c63fdba8d4393b21/adapter_test.go#L421-L428
First use ```initDbResolver``` to get ```dbPool```
then use ``` NewAdapterByMulDb```
If you have multiple DB, you first need to pass in parameters to ```initDbResolver```

Reply:
`initDbResolver` is a private method

Reply:
I have corrected it